File: src_local/agents/Flutter/echo.py
"""
SISTEMA DE PERSONAS AGENTES - NÚCLEO FLUTTER
Módulo: Eco de Observabilidade (Echo)
Função: Auditar rastreabilidade, logs estruturados e tratamento de erros em Dart.
Soberania: ACTIVE-AGENT.
"""
from src_local.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class EchoPersona(BaseActivePersona):
    """
    Core: PhD in System Observability & Real-time Telemetry (Flutter) 🗣️
    
    Responsabilidades:
    1. Rastreabilidade: Garante que o fluxo de execução seja logado.
    2. Gestão de Erros: Detecta blocos catch vazios ou print() genéricos.
    3. Telemetria: Assegura que falhas críticas gerem eventos estruturados.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Echo", "🗣️", "PhD Observability Engineer", "Flutter"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Visibilidade e Rastreabilidade Flutter...")
        
        audit_rules = [
            {'regex': r"print\(", 'issue': 'Aviso: Uso de print() em vez de log() do developer ou logger.', 'severity': 'medium'},
            {'regex': r"catch\s*\(\w+\)\s*\{\s*\}", 'issue': 'Cegueira: Ponto Cego detectado em Dart (except empty).', 'severity': 'critical'}
        ]
        
        results = self.find_patterns(('.dart',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        # O Echo agora delega a auditoria de observabilidade para o AuditEngine via perform_audit
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em telemetria e rastro digital Flutter."

File: src_local/agents/Bun/Content/echo.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditRule, StrategicFinding, AuditFinding } from "../../base.ts";
import type { ProjectContext } from "../../../core/types.ts";

/**
 * 🔊 EchoPersona (Bun Stack) - PhD in Traceability & Native Auditing
 */
export class EchoPersona extends BaseActivePersona {
    constructor(projectRoot: string | undefined = undefined) {
        super(projectRoot);
        this.name = "Echo";
        this.emoji = "🔊";
        this.role = "PhD Audit Engineer";
        this.phd_identity = "High-Fidelity Traceability & Log Integrity";
        this.stack = "Bun";
    }

    override async execute(context: ProjectContext): Promise<AuditFinding[]> {
        this.setContext(context);
        const findings = await this.performAudit();
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.ts', '.js'],
            rules: [
                { regex: /console\.log\(/, issue: 'Rastreabilidade: Use um logger estruturado em vez de console.log.', severity: 'low' }
            ]
        };
    }

    public audit(): any[] { return []; }
    public Branding(): string { return `${this.emoji} ${this.name}`; }
    public Analysis(): string { return "Traceability Analysis Complete"; }
    public test(): boolean {
        this.Branding();
        this.Analysis();
        this.audit();
        return true;
    }

    override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | null {
        return {
            file, severity: "INFO",
            issue: `PhD Echo (Bun): Garantindo rastreabilidade para ${objective}.`,
            context: "analyzing traceability"
        };
    }

    override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [], branding: this.Branding() };
    }

    override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, PhD em rastreabilidade Bun. Status: ${this.Analysis()}`;
    }
}
